### ISSUE TITLE
Isolate overlapping regions

### ISSUE THREAD

**Issue by Robinlovelace** [npct/pct-shiny#88]
Please can you reproduce @nikolai-b or @usr110 ?

I get

```
Listening on http://127.0.0.1:4535
OGR data source with driver: GeoJSON 
Source: "las-pcycle.geojson", layer: "OGRGeoJSON"
with 326 features
It has 9 fields
Already up-to-date.
Warning in min(values) : no non-missing arguments to min; returning Inf
Warning in min(x, na.rm = na.rm) :
  no non-missing arguments to min; returning Inf
Warning in max(x, na.rm = na.rm) :
  no non-missing arguments to max; returning -Inf
Warning: Unhandled error in observer: non-numeric argument to mathematical function
observe({
    leafletProxy("map") %>% clearGroup(., "zones") %>% clearGroup(., 
        "centers") %>% addPolygons(., data = helper$zones, weight = 2, 
        fillOpacity = transpRate(), opacity = 0.2, fillColor = getColourRamp(zcols, 
            helper$zones[[zoneData()]]), color = "black", group = "zones", 
        options = pathOptions(clickable = F)) %>% addCircleMarkers(., 
        data = helper$cents, radius = circleRadius(), color = "black", 
        group = "centers", popup = zonePopup(helper$cents, scenario(), 
            zoneAttr()))
    isolate({
        leafletProxy("map") %>% {
            if (input$line_type == "straight") 
                plotLines(., helper$l, input$nos_lines, straightPopup, 
                  "maroon")
            else .
        } %>% {
            if (input$line_type == "route") 
                leafletProxy("map") %>% plotLines(., helper$rQuiet, 
                  input$nos [... truncated]
```

Not sure what's causing this - finding pct-shiny really hard to debug...

**Comment by usr110** [comment]
@Robinlovelace Please give it a try and see if it is working now. 
On a different note, can you tell me why you revert https://github.com/npct/pct-shiny/commit/3255c1b81097ee62417bfc9ce66f06cb8eeaf91a?
I introduced a more modular approach to define colours for different lines at one place.

**Comment by Robinlovelace** [comment]
Many thanks for the fix Ali, that's fantastic.

I'm afraid I cannot remember why I reverted that commit - probably something to do with tests that @nikolai-b and I were doing to update the network visuals. Happy for you to re-revert it, provided it's tested and looks good - trust your judgement on that.

**Comment by usr110** [comment]
Updated by: https://github.com/npct/pct-shiny/commit/7bb0fddfb63894577b64fa50ed8e8c8441c0b327

**Comment by nikolai-b** [comment]
@Robinlovelace I think you did it by accident in 9f6aeca, maybe something happened when you pull the git repo because it looks like you wiped out 3255c1b.  Nice fix @usr110 btw.

**Comment by Robinlovelace** [comment]
@usr110 it still seems to be broken when I run it on my computer and the server (just pushed to production) - see /West-Yorkshire online!

error seems similar:

```
> runApp('West-Yorkshire')

Listening on http://127.0.0.1:7975
OGR data source with driver: GeoJSON 
Source: "las-pcycle.geojson", layer: "OGRGeoJSON"
with 326 features
It has 9 fields
Already up-to-date.
Warning in min(values) : no non-missing arguments to min; returning Inf
Warning in min(x, na.rm = na.rm) :
  no non-missing arguments to min; returning Inf
Warning in max(x, na.rm = na.rm) :
  no non-missing arguments to max; returning -Inf
Warning: Unhandled error in observer: non-numeric argument to mathematical function
observe({
    leafletProxy("map") %>% clearGroup(., "zones") %>% clearGroup(., 
        "centers") %>% addPolygons(., data = helper$zones, weight = 2, 
        fillOpacity = transpRate(), opacity = 0.2, fillColor = getColourRamp(zcols, 
            helper$zones[[zoneData()]]), color = "black", group = "zones", 
        options = pathOptions(clickable = F)) %>% addCircleMarkers(., 
        data = helper$cents, radius = circleRadius(), color = "black", 
        group = "centers", popup = zonePopup(helper$cents, scenario(), 
            zoneAttr()))
    isolate({
        leafletProxy("map") %>% {
            if (input$line_type == "straight") 
                plotLines(., helper$l, input$nos_lines, straightPopup, 
                  "straight_line", getLineColour("straight_line"))
            else .
        } %>% {
            if (input$line_type == "route") 
                leafletProxy("map") %>% plotLines(., help [... truncated]
```

**Comment by usr110** [comment]
@Robinlovelace It does throw these warnings and stops working in the browser. However, at my end, this app seems to be working fine, as long as it is run within RStudio - even after these warnings. 

It is to do with the overlap boundary of leeds with west-yorkshire. See the image below.

![overlap-la](https://cloud.githubusercontent.com/assets/10958628/11213677/2951d9d4-8d35-11e5-9f5c-fece7b0d51e3.png)

If you go to pct-data folder and then rename leeds folder to leeds1 (or whatever you like), then it doesn't throw any warnings/errors. Try it and then let me know if it is working or not.
So in short, if you like west-yorkshire to work, remove leeds from the pct-data. Overlapping LAs create problems.

**Comment by Robinlovelace** [comment]
Hi @usr110 thanks for this - that indeed does solve the problem.

@nikolai-b any ideas for a higher-level fix? I think we should support multiple regional levels. A simple tag indicating non-standard regions could turn off change of scope for these, right?

Leaving open as this will continue to be a problem for other combined authorities and a higher-level fix will be useful in the long-term.

**Comment by Robinlovelace** [comment]
Links to this: https://github.com/npct/pct-shiny/issues/80 I'm going to push a proposed fix.

**Comment by Robinlovelace** [comment]
Note the 'isolate' file in the data dir to indicate regions that should never jump: https://github.com/npct/pct-data/commit/767bda4a96fadc8fee088fdb29a538fc3702938b

And here's a commit to pct to automate that: https://github.com/npct/pct/commit/66b06066c0cdf9d4faff969af9dbef4c62cd28f0

Make sense @usr110 and @nikolai-b ?

Now we just need `if(!exists("data_dir/isolated")) ...` somewhere in master.R, right?
